src/app/booking/useDotButton.js

```diff
@@ -0,0 +1,58 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import {IconButton} from "@mui/material";
+import {Box} from "@mui/system";
+import {useTheme} from "@mui/material/styles";
+
+export const useDotButton = (emblaApi) => {
+  const [selectedIndex, setSelectedIndex] = useState(0)
+  const [scrollSnaps, setScrollSnaps] = useState([])
+
+  const onDotButtonClick = useCallback(
+    (index) => {
+      if (!emblaApi) return
+      emblaApi.scrollTo(index)
+    },
+    [emblaApi]
+  )
+
+  const onInit = useCallback((emblaApi) => {
+    setScrollSnaps(emblaApi.scrollSnapList())
+  }, [])
+
+  const onSelect = useCallback((emblaApi) => {
+    setSelectedIndex(emblaApi.selectedScrollSnap())
+  }, [])
+
+  useEffect(() => {
+    if (!emblaApi) return
+
+    onInit(emblaApi)
+    onSelect(emblaApi)
+    emblaApi.on('reInit', onInit)
+    emblaApi.on('reInit', onSelect)
+    emblaApi.on('select', onSelect)
+  }, [emblaApi, onInit, onSelect])
+
+  return {
+    selectedIndex,
+    scrollSnaps,
+    onDotButtonClick
+  }
+}
+
+export const DotButton = ({ index, isSelected, onClick }) => {
+  const theme = useTheme();
+
+  return (
+    <IconButton onClick={onClick} size="small" sx={{ padding: '5px', margin: '0 5px' }}>
+      <Box
+        sx={{
```

Comment:
I'd like to keep initial example code here by maximum. And use outer styles passed via props instead for customization.